bucket/archwsl.json

```diff
@@ -43,30 +43,19 @@
         "    $installable = $false",
         "}",
         "if ($installable) {",
-        "    \"`$si = New-Object System.Diagnostics.ProcessStartInfo '$dir\\Arch.exe' -Property @{ UseShellExecute = `$false; RedirectStandardInput = `$true }",
-        "    `$p = [System.Diagnostics.Process]::Start(`$si)",
-        "    `$p.StandardInput.Write('y')",
-        "    `$p.WaitForExit()",
-        "    `$lx_uid = & '$dir\\Arch.exe' get --lxuid",
-        "    if(!`$lx_uid) { error 'ArchWsl is not installed!'; return }",
-        "    Set-ItemProperty -Path ('HKCU:Software\\Microsoft\\Windows\\CurrentVersion\\Lxss\\' + `$lx_uid) -Name BasePath -Value '$persist_dir\\data'",
-        "    if (Test-Path ('$persist_dir' + '\\data\\rootfs')) {",
-        "        Remove-Item -Recurse -Path '$dir\\rootfs'",
-        "    } else {",
-        "        Move-Item -Path '$dir\\rootfs' -Destination '$persist_dir\\data\\rootfs'",
-        "    }",
-        "    Remove-Item -Recurse -Path '$dir\\temp'",
+        "    \"Copy-Item '$dir\\Arch.exe' '$persist_dir\\data\\Arch.exe'",
+        "    `& '$persist_dir\\data\\Arch.exe' install `\"$dir\\rootfs.tar.gz`\";`$res=`$?",
+        "    Remove-Item ('$persist_dir'+'\\data\\Arch.exe') -Force",
+        "    if(!`$res) { error 'ArchWsl installation failed!'; return }",
         "    \" | Set-Content \"$dir\\install.ps1\"",
         "    & \"$dir\\install.ps1\"",
         "}"
     ],
     "uninstaller": {
         "script": [
-            "$lx_uid = & \"$dir\\Arch.exe\" get --lxuid",
-            "if(!$lx_uid) { error 'ArchWsl is not registered!'; return }",
-            "Set-ItemProperty -Path \"HKCU:Software\\Microsoft\\Windows\\CurrentVersion\\Lxss\\$lx_uid\" -Name BasePath -Value $dir",
-            "New-Item -ItemType Directory -Path \"$dir\\rootfs\" | Out-Null",
-            "'y' | & \"$dir\\Arch.exe\" clean"
+            "& \"$dir\\Arch.exe\" isregd",
+            "if(!$?) { error 'ArchWsl is not registered!'; return }",
```

Comment:
`!$?` what's the meaning of this?

Reply:
`Arch.exe isregd` is `True` if registered in the system, otherwise it returns `False`.
`$?` stores the return value (True / False) of the last command.
`!$?` Is a logical inversion of it.